interop/test_utils.go

```diff
@@ -454,6 +454,49 @@ func DoCancelAfterFirstResponse(tc testpb.TestServiceClient) {
 	}
 }
 
+// DoStatusCodeAndMessage checks that the status code is propagated back to the client.
+func DoStatusCodeAndMessage(tc testpb.TestServiceClient) {
+	var code int32 = 2
+	msg := "test status message"
+	respStatus := &testpb.EchoStatus{
+		Code:    &code,
+		Message: &msg,
+	}
+	// Test UnaryCall.
+	req := &testpb.SimpleRequest{
+		ResponseStatus: respStatus,
+	}
+	_, err := tc.UnaryCall(context.Background(), req)
```

Comment:
Can't use that here, because I'm using the value of `err` again on line 473, outside of this `if` statement.

But probably no longer relevant, given the change you suggested below.